crates/llm_gateway/src/stream_context.rs

```diff
@@ -344,6 +420,75 @@ impl HttpContext for StreamContext {
                     }
                 };
             self.response_tokens += token_count;
+
+            // Compute TTFT if not already recorded
+            if !self.ttft_recorded {
```

Comment:
You don't need separate `ttft_recorded`. You could use optional `ttft_duration` for the same purpose